src/components/button-toggle/button-toggle.ts

```diff
@@ -0,0 +1,333 @@
+  private _change: EventEmitter<MdButtonToggleChange> = new EventEmitter<MdButtonToggleChange>();
+  @Output() get change(): Observable<MdButtonToggleChange> {
+    return this._change.asObservable();
+  }
+
+  /** Child button toggle buttons. */
+  @ContentChildren(forwardRef(() => MdButtonToggle))
+  private _buttonToggles: QueryList<MdButtonToggle> = null;
+
+  @Input()
+  get name(): string {
+    return this._name;
+  }
+
+  set name(value: string) {
+    this._name = value;
+
+    this._updateButtonToggleNames();
+  }
+
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value) {
+    this._disabled = (value != null && value !== false) ? true : null;
+  }
+
+  @Input()
+  get value(): any {
+    return this._value;
+  }
+
+  set value(newValue: any) {
+    if (this._value != newValue) {
+      this._value = newValue;
+
+      this._updateSelectedButtonToggleFromValue();
+      this._emitChangeEvent();
+    }
+  }
+
+  @Input()
+  get selected() {
+    return this._selected;
+  }
+
+  set selected(selected: MdButtonToggle) {
+    this._selected = selected;
+    this.value = selected ? selected.value : null;
+
+    if (selected && !selected.checked) {
+      selected.checked = true;
+    }
+  }
+
+  private _updateButtonToggleNames(): void {
+    (this._buttonToggles || []).forEach((toggle) => {
+      toggle.name = this._name;
+    });
+  }
+
+  // TODO: Refactor into shared code with radio.
+  private _updateSelectedButtonToggleFromValue(): void {
+    let isAlreadySelected = this._selected != null && this._selected.value == this._value;
+
+    if (this._buttonToggles != null && !isAlreadySelected) {
+      let matchingButtonToggle = this._buttonToggles.filter(
+          buttonToggle => buttonToggle.value == this._value)[0];
+
+      if (matchingButtonToggle) {
+        this.selected = matchingButtonToggle;
+      } else if (this.value == null) {
+        this.selected = null;
+        this._buttonToggles.forEach(buttonToggle => {buttonToggle.checked = false; });
+      }
+    }
+  }
+
+  /** Dispatch change event with current selection and group value. */
+  private _emitChangeEvent(): void {
+    let event = new MdButtonToggleChange();
+    event.source = this._selected;
+    event.value = this._value;
+    this._change.emit(event);
+  }
+}
+
+/** Multiple selection button-toggle group. */
+@Directive({
+  selector: 'md-button-toggle-group[multiple]',
+})
+export class MdButtonToggleGroupMultiple {
+  /** Disables all toggles in the group. */
+  private _disabled: boolean = null;
+
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value) {
+    this._disabled = (value != null && value !== false) ? true : null;
+  }
+}
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-button-toggle',
+  templateUrl: 'button-toggle.html',
+  styleUrls: ['button-toggle.css'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class MdButtonToggle implements OnInit {
```

Comment:
seems to me the toggle doesn't need a dependency to its group and that would simplify its logic. The group could `ContentChildren` and listen to change events and coordinate its own state.

It feels weird that the toggle is also responsible for changing the state of the group, and by clearing up responsibility you could definitely set `ChangeDetectionStrategy` to `OnPush`.